src/thunderbots/software/ai/navigator/path_planner/astar.h

```diff
@@ -0,0 +1,149 @@
+#pragma once
+
+// I apologize if your CLion slows to a crawl
+#include <boost/graph/astar_search.hpp>
+#include <boost/graph/grid_graph.hpp>
+
+#include "ai/world/field.h"
+#include "path_planner.h"
+namespace AStar
+{
+    // TODO: potentially move a lot of stuff out of this header
+
+    // number of graph vertices per metre
+    // TODO: make this a dynamic parameter or passed in the constructor
+    // to AStarGridGraph
+    static constexpr unsigned long GRID_VERTEX_DENSITY = 20;
```

Comment:
I think that in the case of this we should pass all the parameters to the constructor, rather then via dynamic parameters if given the choice. This will make it easier to test (less worry about side effects), and easier to write, and besides, I don't see us wanting to change options like this in the middle of path-finding.

Reply:
fixed